src/main/scala/coupledL2/tl2chi/MainPipe.scala

```diff
@@ -933,4 +934,16 @@ class MainPipe(implicit p: Parameters) extends TL2CHIL2Module with HasCHIOpcodes
 
   XSPerfAccumulate("early_prefetch", meta_s3.prefetch.getOrElse(false.B) && !meta_s3.accessed && !dirResult_s3.hit && task_s3.valid)
 
+  /* ===== Hardware Performance Monitor ===== */
+  val perfEvents = Seq(
+    ("l2_cache_access", task_s3.valid && (sinkA_req_s3 && !req_prefetch_s3 || sinkC_req_s3)),
+    ("l2_cache_wb", task_s3.valid && (mshr_cbWrData_s3 || mshr_snpRespDataX_s3)),
+    ("l2_cache_allocate", task_s3.valid && sinkC_req_s3 && (req_s3.opcode === ReleaseData)),
```

Comment:
What does `l2_cache_allocate` mean?

Reply:
Actually this name is from ARM's manual.😂
It means L2 data cache allocation without refill. I think this name is confusing, too. We can rename this event.